src/components/TaskList.tsx

```diff
@@ -22,12 +22,28 @@ export default TaskList;
 
 function ExpandableTaskItem({ item }: { item: TaskItem }) {
   const [isExpanded, setIsExpanded] = useState(false);
+  const [isChecked, setIsChecked] = useState(false);
 
   // function which will not be re-created for each re-rendering
   const handleClick = useCallback(
     () => setIsExpanded((expanded) => !expanded),
     []
   );
+  const handleCheckboxChange = useCallback(() => {
+    setIsChecked((checked) => !checked);
+  }, []);
+
+  useEffect(() => {
+    const checkbox = window.localStorage.getItem(`TASK_CHECKED:${item.id}`);
+    if (checkbox !== null) setIsChecked(JSON.parse(checkbox as any));
+  }, []);
```

Comment:
Is this correct?

This will run only once when the component is mounted the first time.

Reply:
Rule of thumb: When a component has more than one `useEffect`, its probably wrong 😆

Reply:
💡 Maybe you can load the item value directly on the `useState` input